experiments/generic/container-kill/crictl.go

```diff
@@ -0,0 +1,209 @@
+
+	//Getting kubeConfig and Generate ClientSets
+	if err := clients.GenerateClientSetFromKubeConfig(); err != nil {
+		log.Fatalf("Unable to Get the kubeconfig due to %v", err)
+	}
+
+	//Fetching all the ENV passed for the runner pod
+	log.Info("[PreReq]: Getting the ENV variables")
+	environment.GetENV(&experimentsDetails)
+
+	//Obtain the pod ID of the application pod
+	podID, err := GetPodID(&experimentsDetails)
+	if err != nil {
+		log.Fatalf("Unable to get the pod id %v", err)
+	}
+
+	log.Infof("Killing the containers of pod with PodID: %v", podID)
+
+	err = KillContainer(&experimentsDetails, clients, podID, &eventsDetails)
+	if err != nil {
+		log.Fatalf("container-kill chaos terminated due to %v", err)
+	}
+
+}
+
+// KillContainer kills the application container
+func KillContainer(experimentsDetails *types.ExperimentDetails, clients environment.ClientSets, podID string, eventsDetails *types.EventDetails) error {
+
+	ChaosStartTimeStamp := time.Now().Unix()
+
+	for iteration := 0; iteration < experimentsDetails.Iterations; iteration++ {
+
+		//Obtain the container ID through Pod
+		containerID, err := GetContainerID(experimentsDetails, podID)
+		if err != nil {
+			return errors.Errorf("Unable to get the container id, %v", err)
+		}
+		log.Infof("Killing the container with containerID: %v", containerID)
+
+		// killing the application container
+		StopContainer(containerID)
+
+		if experimentsDetails.EngineName != "" {
+			msg := "Injecting " + experimentsDetails.ExperimentName + " chaos on application pod"
+			environment.SetEventAttributes(eventsDetails, types.ChaosInject, msg)
+			events.GenerateEvents(experimentsDetails, clients, eventsDetails)
+		}
+
+		//Waiting for the chaos interval after chaos injection
+		if experimentsDetails.ChaosInterval != 0 {
+			log.Infof("[Wait]: Wait for the chaos interval %vs", strconv.Itoa(experimentsDetails.ChaosInterval))
+			waitForChaosInterval(experimentsDetails)
+		}
+
+		//Check the status of restarted container
+		err = CheckContainerStatus(experimentsDetails, clients)
+		if err != nil {
+			return errors.Errorf("Application container is not running, %v", err)
+		}
+
+		ChaosCurrentTimeStamp := time.Now().Unix()
+		chaosDiffTimeStamp := ChaosCurrentTimeStamp - ChaosStartTimeStamp
+
+		// terminating the execution after the timestamp exceed the total chaos duration
+		if int(chaosDiffTimeStamp) >= experimentsDetails.ChaosDuration {
+			break
+		}
+
+	}
+	log.Infof("[Completion]: %v chaos is done", experimentsDetails.ExperimentName)
+	return nil
+
+}
+
+//GetPodID derive the pod-id of the application pod
+func GetPodID(experimentsDetails *types.ExperimentDetails) (string, error) {
+
+	cmd := exec.Command("crictl", "pods")
+	stdout, _ := cmd.Output()
+
+	pods := removeExtraSpaces(stdout)
+	for i := 0; i < len(pods)-1; i++ {
+		attributes := strings.Split(pods[i], " ")
+		// fmt.Printf("podlist: %v", attributes)
+		if attributes[3] == experimentsDetails.ApplicationPod {
```

Comment:
can we just check against the latest crictl version once?

Reply:
tried with 1.18.0(latest) working fine